### DISCUSSION TITLE
Playback durations are wrong.  Need troubleshooting help

### DISCUSSION THREAD

**Discussion by troycarpenter** [ampache/ampache#4001] [Q&A] [answered]
I've started getting comments from my users about songs abruptly ending before they should.  During the investigation, I'm finding some oddities.  At first I though it was with the mobile player Dsub, but now I'm suspecting the server since I've seen the same issue on web interface.

I'll start with the Web interface...Here's an example where a song is truncated:

![image](https://github.com/user-attachments/assets/ebb78009-83a9-4090-bff7-de62544c7a16)
You can see the player thinks the length is 3:23.  However, the playback indicator shows it's at the end of the song with about 20 seconds still to go.  Even the waveform bar (the red one) shows it's at the end.

For the same song In Dsub the length of the song shows correctly in the album list, but the player shows the end of the song is 3:01 just as the web player seems to.   In this case, the song will cut off before it is finished.
![dsub1](https://github.com/user-attachments/assets/ab022156-ea77-4abc-b7a6-e2c880ded7db)
To triple check,  I've seen the same behavior in Ample.  I've verified the file is correct:
![image](https://github.com/user-attachments/assets/b92bf1a7-8eae-445a-a147-44976f0bef01)

Below is an example of this phenomena happening in the other direction.  Again, the player correctly shows the song length as 3:58, but in the screen shot you can see the playback is almost there but shows over two minutes left in the song.  The player does stop playing at the proper end of the song but the indicator instantly jumps over the last two minutes of display to the end.

![image](https://github.com/user-attachments/assets/0eafebe8-3a29-4edf-8747-36238ace4bb7)
In Dsub, the same thing.  The song says it's 3:58, but the player thinks 5:35.  In this case the song will end normally and move the next song.
![Screenshot_20240802_120139_DSub](https://github.com/user-attachments/assets/d281a160-d7a0-4736-898f-258a2f2cd022)
![image](https://github.com/user-attachments/assets/9dfab245-f1b9-451f-a61e-8496055407d0)

Not all songs are doing this.  I'm thinking something in the database or cache somewhere (although I do have transcoding turned off).  I find it interesting the waveform element seems to follow the "incorrect" lengths as well.

Is anyone else experiencing this?  Have I missed something simple?

**Comment by lachlan-00** [comment]
can you use the dev tools to see what kind of responses you're getting and the headers
![image](https://github.com/user-attachments/assets/ba98e54c-e065-48f1-a0a1-6788c543ec70)

especially if it's giving you the x-content duration
![image](https://github.com/user-attachments/assets/012bf1e4-b785-432b-bbff-8f1785ec1310)


when you're not transcoding it should just be returning the time in the database

**Reply by troycarpenter** [reply]
Where do you see those headers?  I've looked through most of the pages I can find, but I've yet to see anything like what you've showed above.

**Comment by troycarpenter** [comment]
As replied on the other thread, 
The reported duration seems correct:
![image](https://github.com/user-attachments/assets/d0ba78b8-4c9c-4270-820a-9adbd3706412)
But when that song is played in any player, it only plays until 3:01, not the expected 3:23.
![image](https://github.com/user-attachments/assets/c1fab93e-902b-4527-9d87-314881a8c74b)

I also see this problem in the opposite direction.  For example:
![image](https://github.com/user-attachments/assets/80bb877c-c445-4f97-b8a9-1c2ad00bcd35)
But in the player I see this:
![image](https://github.com/user-attachments/assets/9390aced-c2b2-4b09-8fb2-67f9319dc6ee)
The playback marker is almost at 3:58, but you can see it's far away from the graphical end of the song.

**Reply by troycarpenter** [reply]
And this is happening on any player directly tied to Ampache that I have tried.  The web player, Dsub and Ample.

**Comment by troycarpenter** [comment]
Interesting.  I had the log running in a terminal after adding a new album.  When I added the album to my mobile app, lo and behold I got a transcode log for every song.

I have this set in the configuration:
![image](https://github.com/user-attachments/assets/0689e4ff-bae7-4cf3-9819-1b917802cb51)
However, I see this in the log:
```
2024-08-16T21:24:17+00:00 [emailaddress@home] (Ampache\Module\Application\Playback\PlayAction) -> Transcoding due to always
```
Now I'm looking through all the settings and in the config file to see what it thinks I need to transcode.

**Reply by troycarpenter** [reply]
My first guess is that the creation of the waveform is generating those logs.

**Reply by troycarpenter** [reply]
The waveform generation seems to be the issue.  When I turned that off in the config file, I added a new album that never had waveforms created.  There were no transcode logs as the songs were downloaded to my mobile device, and now all song durations in the player match the song duration in the long list.  For instance:
![image](https://github.com/user-attachments/assets/153132e2-3ada-4d01-bee5-080268275481)
You can see that the last song in the playlist (which is currently playing in the player) has a duration of 3:47, which matches what the player shows.  Also, when I play the song to the end, the song is complete.  Not cut off nor showing more time to play.

So it is my belief that the transcode which is happening to create the graphical waveform is causing the issue, and the transcoded song is being sent to the player with a wrong duration.

My next question will obviously be how to get rid of these transcoded files.

Also, can this be independently verified before I turn this into an bug?

**Reply by troycarpenter** [reply]
On my mobile phone, simply deleting any cached songs fixed the duration problem.  The first song listed in this thread "Third Rate Romance", which had a duration of 3:23 in the database, but only 3:01 in the player, now shows correctly, and plays to the end.

The other issue I brought up in #4002 also went away.  With the waveform removed,  a simple bar showed up instead and I was able to jump to any point in the song.

I will write this up in a separate answer so I can mark it as such.

**Reply by lachlan-00** [reply]
waveform, now that's interesting, there must be something for that that's ignoring the length from the DB

**Comment by troycarpenter** [comment, marked as answer]
The problem described arose from this setting in the configuration file:
```
; Waveform
; This settings tells Ampache to attempt to generate a waveform
; for each song. It requires transcode and encode_args_wav settings.
; You must also set tmp_dir_path in order for this to work
; DEFAULT: "false"
waveform = "false"
```
When that is set to true, it forces transcoding and apparently that transcoded song is sent to the player instead of the original.  This transcoded song has the wrong durations and messes with what the players think the duration actually is.

By turning this setting off, and deleting any player cache that may exist, the song duration problem is fixed.  I don't know if the waveform feature should be working this way, but it does seem to be broken if enabled.  This is the same solution for #4002.
